p22/tests/benchmark_test.rs

```diff
@@ -0,0 +1,31 @@
+#![feature(test)]
+extern crate test;
+#[cfg(test)]
+mod tests {
```

Comment:
You do not need to use `#[cfg(test)]` in integration tests and benchmarks (since it always will be true). Also you do not need to introduce `mod tests` for them.